src/sentry/workflow_engine/migrations/0032_migrate_metric_alerts.py

```diff
@@ -0,0 +1,468 @@
+    Incident = apps.get_model("sentry", "Incident")
+
+    Action = apps.get_model("workflow_engine", "Action")
+    ActionAlertRuleTriggerAction = apps.get_model("workflow_engine", "ActionAlertRuleTriggerAction")
+    AlertRuleDetector = apps.get_model("workflow_engine", "AlertRuleDetector")
+    AlertRuleWorkflow = apps.get_model("workflow_engine", "AlertRuleWorkflow")
+    DataCondition = apps.get_model("workflow_engine", "DataCondition")
+    DataConditionGroup = apps.get_model("workflow_engine", "DataConditionGroup")
+    DataConditionGroupAction = apps.get_model("workflow_engine", "DataConditionGroupAction")
+    DataSource = apps.get_model("workflow_engine", "DataSource")
+    Detector = apps.get_model("workflow_engine", "Detector")
+    DetectorState = apps.get_model("workflow_engine", "DetectorState")
+    DetectorWorkflow = apps.get_model("workflow_engine", "DetectorWorkflow")
+    Workflow = apps.get_model("workflow_engine", "Workflow")
+    WorkflowDataConditionGroup = apps.get_model("workflow_engine", "WorkflowDataConditionGroup")
+
+    for alert_rule in RangeQuerySetWrapperWithProgressBarApprox(
+        AlertRule.objects_with_snapshots.all()
+    ):
+        if alert_rule.status in [AlertRuleStatus.DISABLED, AlertRuleStatus.SNAPSHOT]:
+            continue
+        if alert_rule.detection_type == "static":
+            logger.info(
+                "anomaly detection alert rule, skipping", extra={"alert_rule_id": alert_rule.id}
+            )
+            continue
+
+        organization_id = alert_rule.organization_id
+        project = alert_rule.projects.get()
+        snoozed = None
+        try:
+            snoozed = RuleSnooze.objects.get(alert_rule_id=alert_rule.id, user_id=None)
+        except RuleSnooze.DoesNotExist:
+            pass
+        enabled = True if snoozed is not None else False
+
+        create_activity = AlertRuleActivity.objects.get(
+            alert_rule_id=alert_rule.id, type=AlertRuleActivityType.CREATED.value
+        )
+
+        # create data source
+        snuba_query = alert_rule.snuba_query
+        if not snuba_query:
+            logger.info("alert rule missing snuba query", extra={"alert_rule_id": alert_rule.id})
+            continue
+        try:
+            query_subscription = QuerySubscription.objects.get(snuba_query=snuba_query.id)
+        except QuerySubscription.DoesNotExist:
+            logger.info(
+                "query subscription does not exist", extra={"snuba_query_id": snuba_query.id}
+            )
+            continue
+        data_source = DataSource.objects.create_or_update(
+            organization_id=organization_id,
+            source_id=str(query_subscription.id),
+            type="snuba_query_subscription",
+        )
+
+        # create DCG
+        data_condition_group = DataConditionGroup.objects.create_or_update(
+            organization_id=organization_id,
+        )
+        # create detector
+        detector = Detector.objects.create_or_update(
```

Comment:
`create_or_update` returns the object and a boolean of whether or not it was created, if you don't need the boolean you can just put `_` e.g. `detector, _ = Detector...`